File: source/kCura.Relativity.Export.ConvertedToC#/Process/ExportSearchProcess.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
using kCura.Relativity.Export.Exports;
using kCura.Relativity.Export.FileObjects;
using kCura.Relativity.Export.Types;

namespace kCura.Relativity.Export.Process
{

	public class ExportSearchProcess : kCura.Windows.Process.ProcessBase
	{

		public ExportFile ExportFile;
		private Exporter withEventsField__searchExporter;
		private Exporter _searchExporter {
			get { return withEventsField__searchExporter; }
			set {
				if (withEventsField__searchExporter != null) {
					withEventsField__searchExporter.FileTransferModeChangeEvent -= _searchExporter_FileTransferModeChangeEvent;
					withEventsField__searchExporter.StatusMessage -= _productionExporter_StatusMessage;
					withEventsField__searchExporter.FatalErrorEvent -= _productionExporter_FatalErrorEvent;
					withEventsField__searchExporter.ShutdownEvent -= _searchExporter_ShutdownEvent;
				}
				withEventsField__searchExporter = value;
				if (withEventsField__searchExporter != null) {
					withEventsField__searchExporter.FileTransferModeChangeEvent += _searchExporter_FileTransferModeChangeEvent;
					withEventsField__searchExporter.StatusMessage += _productionExporter_StatusMessage;
					withEventsField__searchExporter.FatalErrorEvent += _productionExporter_FatalErrorEvent;
					withEventsField__searchExporter.ShutdownEvent += _searchExporter_ShutdownEvent;
				}
			}
		}
		private System.DateTime _startTime;
		private Int32 _errorCount;
		private Int32 _warningCount;

		private string _uploadModeText = null;
		protected override void Execute()
		{
			_startTime = DateTime.Now;
			_warningCount = 0;
			_errorCount = 0;
			_searchExporter = new Exporter(this.ExportFile, this.ProcessController);

			if (!_searchExporter.ExportSearch()) {
				this.ProcessObserver.RaiseProcessCompleteEvent(false, _searchExporter.ErrorLogFileName);
			} else {
				this.ProcessObserver.RaiseStatusEvent("", "Export completed");
				this.ProcessObserver.RaiseProcessCompleteEvent();
			}
		}

		private void _searchExporter_FileTransferModeChangeEvent(string mode)
		{
			if (_uploadModeText == null) {
				_uploadModeText = Config.FileTransferModeExplanationText(false);
			}
			this.ProcessObserver.RaiseStatusBarEvent("File Transfer Mode: " + mode, _uploadModeText);
		}

		private void _productionExporter_StatusMessage(ExportEventArgs e)
		{
			switch (e.EventType) {
				case kCura.Windows.Process.EventType.Error:
					_errorCount += 1;
					this.ProcessObserver.RaiseErrorEvent(e.DocumentsExported.ToString(), e.Message);
					break;
				case kCura.Windows.Process.EventType.Progress:
					this.ProcessObserver.RaiseStatusEvent("", e.Message);
					break;
				case kCura.Windows.Process.EventType.Status:
					this.ProcessObserver.RaiseStatusEvent(e.DocumentsExported.ToString(), e.Message);
					break;
				case kCura.Windows.Process.EventType.Warning:
					_warningCount += 1;
					this.ProcessObserver.RaiseWarningEvent(e.DocumentsExported.ToString(), e.Message);
					break;
			}
			IDictionary statDict = null;
			if ((e.AdditionalInfo != null) && e.AdditionalInfo is IDictionary) {
				statDict = (IDictionary)e.AdditionalInfo;
			}

			this.ProcessObserver.RaiseProgressEvent(e.TotalDocuments, e.DocumentsExported, _warningCount, _errorCount, _startTime, new DateTime(), null, null, statDict);
		}

		private void _productionExporter_FatalErrorEvent(string message, System.Exception ex)
		{
			this.ProcessObserver.RaiseFatalExceptionEvent(ex);
		}

		private void _searchExporter_ShutdownEvent()
		{
			this.ProcessObserver.Shutdown();
		}
	}
}

File: source/kCura.Relativity.Export/Process/ExportSearchProcess.vb
Imports kCura.Relativity.Export.Exports
Imports kCura.Relativity.Export.FileObjects
Imports kCura.Relativity.Export.Types

Namespace kCura.Relativity.Export.Process

	Public Class ExportSearchProcess
		Inherits kCura.Windows.Process.ProcessBase

		Public ExportFile As ExportFile
		Private WithEvents _searchExporter As Exporter
		Private _startTime As System.DateTime
		Private _errorCount As Int32
		Private _warningCount As Int32
		Private _uploadModeText As String = Nothing

		Protected Overrides Sub Execute()
			_startTime = DateTime.Now
			_warningCount = 0
			_errorCount = 0
			_searchExporter = New Exporter(Me.ExportFile, Me.ProcessController)

			If Not _searchExporter.ExportSearch() Then
				Me.ProcessObserver.RaiseProcessCompleteEvent(False, _searchExporter.ErrorLogFileName)
			Else
				Me.ProcessObserver.RaiseStatusEvent("", "Export completed")
				Me.ProcessObserver.RaiseProcessCompleteEvent()
			End If
		End Sub

		Private Sub _searchExporter_FileTransferModeChangeEvent(ByVal mode As String) Handles _searchExporter.FileTransferModeChangeEvent
			If _uploadModeText Is Nothing Then
				_uploadModeText = Config.FileTransferModeExplanationText(False)
			End If
			Me.ProcessObserver.RaiseStatusBarEvent("File Transfer Mode: " & mode, _uploadModeText)
		End Sub

		Private Sub _productionExporter_StatusMessage(ByVal e As ExportEventArgs) Handles _searchExporter.StatusMessage
			Select Case e.EventType
				Case kCura.Windows.Process.EventType.Error
					_errorCount += 1
					Me.ProcessObserver.RaiseErrorEvent(e.DocumentsExported.ToString, e.Message)
				Case kCura.Windows.Process.EventType.Progress
					Me.ProcessObserver.RaiseStatusEvent("", e.Message)
				Case kCura.Windows.Process.EventType.Status
					Me.ProcessObserver.RaiseStatusEvent(e.DocumentsExported.ToString, e.Message)
				Case kCura.Windows.Process.EventType.Warning
					_warningCount += 1
					Me.ProcessObserver.RaiseWarningEvent(e.DocumentsExported.ToString, e.Message)
			End Select
			Dim statDict As IDictionary = Nothing
			If Not e.AdditionalInfo Is Nothing AndAlso TypeOf e.AdditionalInfo Is IDictionary Then
				statDict = DirectCast(e.AdditionalInfo, IDictionary)
			End If

			Me.ProcessObserver.RaiseProgressEvent(e.TotalDocuments, e.DocumentsExported, _warningCount, _errorCount, _startTime, New DateTime, Nothing, Nothing, statDict)
		End Sub

		Private Sub _productionExporter_FatalErrorEvent(ByVal message As String, ByVal ex As System.Exception) Handles _searchExporter.FatalErrorEvent
			Me.ProcessObserver.RaiseFatalExceptionEvent(ex)
		End Sub

		Private Sub _searchExporter_ShutdownEvent() Handles _searchExporter.ShutdownEvent
			Me.ProcessObserver.Shutdown()
		End Sub
	End Class
End Namespace